src/main/java/fr/arthurbambou/fdlink/discordstuff/DiscordBot.java

```diff
@@ -86,40 +85,44 @@ public DiscordBot(String token, FDLink.Config config) {
 
-        if (this.config.minecraftToDiscord.booleans.serverStartingMessage) {
+        if (this.config.minecraftToDiscord.chatChannels.serverStartingMessage || this.config.minecraftToDiscord.logChannels.serverStartingMessage) {
             ServerLifecycleEvents.SERVER_STARTING.register(minecraftServer -> {
-                sendToAllChannels(this.config.minecraftToDiscord.messages.serverStarting);
+                if (this.config.minecraftToDiscord.chatChannels.serverStartingMessage) sendToChatChannels(config.minecraftToDiscord.messages.serverStarting);
+                if (this.config.minecraftToDiscord.logChannels.serverStartingMessage) sendToLogChannels(config.minecraftToDiscord.messages.serverStarting);
             });
         }
 
-        if (this.config.minecraftToDiscord.booleans.serverStartMessage) {
+        if (this.config.minecraftToDiscord.chatChannels.serverStartMessage || this.config.minecraftToDiscord.logChannels.serverStartMessage) {
             ServerLifecycleEvents.SERVER_STARTED.register((server -> {
                 startTime = server.getServerStartTime();
-                sendToAllChannels(this.config.minecraftToDiscord.messages.serverStarted);
+                if (this.config.minecraftToDiscord.chatChannels.serverStartMessage) sendToChatChannels(config.minecraftToDiscord.messages.serverStarted);
+                if (this.config.minecraftToDiscord.logChannels.serverStartMessage) sendToLogChannels(config.minecraftToDiscord.messages.serverStarted);
             }));
         }
         ServerLifecycleEvents.SERVER_STOPPING.register(minecraftServer -> {
             this.api.removeListener(MessageCreateListener.class, messageCreateListener);
-            if (this.config.minecraftToDiscord.booleans.serverStoppingMessage) sendToAllChannels(config.minecraftToDiscord.messages.serverStopping);
+            if (this.config.minecraftToDiscord.chatChannels.serverStoppingMessage) sendToChatChannels(config.minecraftToDiscord.messages.serverStopping);
+            if (this.config.minecraftToDiscord.logChannels.serverStoppingMessage) sendToLogChannels(config.minecraftToDiscord.messages.serverStopping);
         });
         ServerLifecycleEvents.SERVER_STOPPED.register((server -> {
-            if (this.config.minecraftToDiscord.booleans.serverStopMessage) {
-                List<CompletableFuture<Message>> requests = sendToAllChannels(config.minecraftToDiscord.messages.serverStopped);
+            if (this.config.minecraftToDiscord.chatChannels.serverStopMessage || this.config.minecraftToDiscord.logChannels.serverStopMessage) {
+                List<CompletableFuture<Message>> requests = new ArrayList<>();
+                if(this.config.minecraftToDiscord.chatChannels.serverStopMessage) requests.add(sendToChatChannels(config.minecraftToDiscord.messages.serverStopped));
+                if(this.config.minecraftToDiscord.logChannels.serverStopMessage) requests.add(sendToLogChannels(config.minecraftToDiscord.messages.serverStopped));
```

Comment:
This added line produces a `NullPointerException` when shutting down the server.
When I comment it out, the crash doesn't happen anymore.

```
ERROR MinecraftServer Exception stopping the server
 java.lang.NullPointerException
	at fr.arthurbambou.fdlink.discordstuff.DiscordBot.lambda$new$5(DiscordBot.java:113)
	at net.minecraft.server.MinecraftServer.handler$zzp000$afterShutdownServer(MinecraftServer.java:1838)
	at net.minecraft.server.MinecraftServer.shutdown(MinecraftServer.java:618)
	at net.minecraft.server.dedicated.MinecraftDedicatedServer.shutdown(MinecraftDedicatedServer.java:500)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:699)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:226)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
Yeah, I see why that would happen. Thanks for the heads up.

Reply:
Is this from the minecraft logs?

Reply:
This is straight from the _server console_. Happens every time for me when running `/stop`, even if run just after startup.
I haven't changed the default config, only added `token` and `chatChannels`. I have no logChannels.
Only other mod on the server is `fabric-api-0.16.2+build.385-1.16.1.jar`.

Currently there is nothing more I can say, however if I comment the line out and recompile, the error doesn't occur anymore.

Reply:
@v-tron do you set the log channel’s booleans to false if there is no log channel in the config?

Reply:
@arthurbambou oh yea, I forgot to do an actual check for hasLogChannels because I thought that the sendToLogChannels function does that. But it actually returns null on no log Channels so that causes the exception